Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/PlayerInput.cs

```diff
@@ -1930,6 +1927,31 @@ private void OnDeviceChange(InputDevice device, InputDeviceChange change)
             }
         }
 
+        private bool IsCurrentControlScheme(ref InputControlScheme scheme, InputDevice[] devices)
+        {
+            var isSameScheme = user.controlScheme.HasValue && user.controlScheme.Value.Equals(scheme);
+            return (isSameScheme &&
+                ArrayHelpers.EqualSets(user.pairedDevices.ToArray(), devices) &&
+                !user.hasMissingRequiredDevices);
+        }
+
+        private void SwitchControlSchemeInternalUnchecked(ref InputControlScheme controlScheme, params InputDevice[] devices)
+        {
+            // Precondition: Control scheme and devices are correlated
```

Comment:
Why not Debug Assert your preconditions?

Reply:
Very valid point, I will check if this is possible, it will be a quite costly assert though which is indirectly implied by the calling code, but can see if its feasible. Guess it could be an assert calling a static method that makes sure all devices are applicable for the control scheme and that they are actually available.

Reply:
Ended up with relaxing this to only require devices to not be null, replaced with assert and removed "Unchecked" part.